.devcontainer/Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM mcr.microsoft.com/vscode/devcontainers/base:bullseye
+
+# Required nix install dependencies.
+RUN apt-get update && apt-get install -y curl bzip2 adduser direnv
+RUN su - vscode -c 'sh <(curl -L https://releases.nixos.org/nix/nix-2.26.3/install) --no-daemon'
```

Comment:
If you use determinate nix (https://github.com/DeterminateSystems/nix-installer), I think you can skip all the experimental flakes stuff. Determinate's flavor enables it by default.

Reply:
I would still need the other two. I looked at the determinate systems installer for a bit, it seems nice. But I've only ever used the upstream installer, and I got it all working (and even pinned). Unless there's a hard reason, I'd rather not switch over just to drop the one line setting the configuration :)